docs/extend-EBS-size.md

```diff
@@ -0,0 +1,62 @@
+# Extending Gluster volume size to accmodate EBS volume resize
+
+This documents details the operational procedure to expand a gluster volume on
+AWS, whose bricks are backed by an EBS instance. Further, the instructions
+relate to bricks and volumes setup using the ansible plays in this repository
+and do not form a general purpose instruction set for any Gluster instance.
+
+It is assumed that gluster brick deamon is up and running, while this exansion
+is in progress (IOW, this is an online expansion procedure).
+
+1. Expand the EBS instance
+    - There are no specific instructions here, AWS website links are the source
+  of truth [1]
+
+2. Pre-flight checks before expanding the backing PV for the EBS disk
+    - Check size of disk: `sudo fdisk -l <dev-path-to-disk>`
+    - Check pv size of containing disk: `sudo pvs <dev-path-to-disk>`
+    - If sizes differ, and PV size is smaller than the disk size, proceed!
+    - Note the existing VG size `sudo vgs <vg-name>`
+
+3. Resize the PV
+    - `sudo pvresize <dev-path-to-disk>`
+    - Check new PV size, `sudo pvs <dev-path-to-disk>` and cross check with
+    earlier reported disk size
+    - Check the new VG size, `sudo vgs <vg-name>` and cross check with PV size
+
+4. Extend the thin-pool
+    - Check the current thin pool size `sudo lvs <vg-name>/pool`
+    - If LV size is smaller than the new VG size, proceed
+    - `sudo lvextend -l 100%VG /dev/mapper/<vg-name>-pool`
+    **NOTE**: We extend the thin pool to occupy entire VG as there is a single
+    thin pool on a given VG in the setup
+    - Recheck LV size `lvs <vg-name>/pool` to ensure it has expanded
+
+5. Extend the thin LV backing the XFS instance
```

Comment:
Let's break this into 2 different procedures:
- Enlarging the thin pool (1-4)
- Enlarging the active file system (5-8)

I suggest this because the overwrite rate will determine the appropriate relative size of active to total, and if that changes, we'll need to adjust them separately.

Reply:
Ack